src/ed25519.rs

```diff
@@ -431,15 +433,47 @@ impl<'d> Deserialize<'d> for Keypair {
             where
                 E: SerdeError,
             {
+                if bytes.len() != KEYPAIR_LENGTH {
+                    return Err(SerdeError::invalid_length(bytes.len(), &self));
+                }
+
                 let secret_key = SecretKey::from_bytes(&bytes[..SECRET_KEY_LENGTH]);
                 let public_key = PublicKey::from_bytes(&bytes[SECRET_KEY_LENGTH..]);
 
-                if secret_key.is_ok() && public_key.is_ok() {
-                    Ok(Keypair{ secret: secret_key.unwrap(), public: public_key.unwrap() })
+                if let (Ok(secret), Ok(public)) = (secret_key, public_key) {
```

Comment:
Refactored to remove the unwrap below, let me know if you would like this cleanup reverted.